src/Presentation/Controllers/UserController.cs

```diff
@@ -0,0 +1,64 @@
+using Microsoft.AspNetCore.Mvc;
+using ExpenseTrackerGrupo4.src.Domain.Entities;
+using ExpenseTrackerGrupo4.src.Infrastructure.Interfaces;
+using ExpenseTrackerGrupo4.src.Infrastructure.Repositories;
+using ExpenseTrackerGrupo4.src.Aplication.Interfaces;
+
+namespace ExpenseTrackerGrupo4.src.Presentation.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class UserController(IUserService userService) : ControllerBase
+{
+    private readonly IUserService _userService = userService;
+
+    [HttpGet("{id}")]
+    public async Task<IActionResult> GetUserById(Guid id)
+    {
+        var user = await _userService.GetUserByIdAsync(id);
+
+        if(user == null)
+        {
+            return NotFound();
+        }
+
+        return Ok();
+    }
+
+    [HttpPut]
+    public async Task<IActionResult> UpdateUser([FromBody] User user)
+    {
+        if(user == null)
+        {
+            return BadRequest();
+        }
+        
+        await _userService.UpdateUserAsync(user);
+        return Ok();
```

Comment:
When updating the data, the password is not hashed again.
![image](https://github.com/user-attachments/assets/23ef5a38-757a-4d9b-8b0f-245d254a8e80)